extensions/aris/src/bigDataCluster/controller/controller.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { EndpointRouterApi } from './apiGenerated';
+import { IEndPointsResponse, IControllerError, IEndPointsRequest, IHttpResponse, IEndPoint } from './types';
+
+export class ArisController {
+
+	private static adjustUrl(url: string): string {
+		if (!url) {
+			return undefined;
+		}
+
+		url = url.trim().replace(/ /g, '').replace(/,(\d+)$/, ':$1');
+		if (!url.includes('://')) {
+			url = `https://${url}`;
+		}
+		return url;
+	}
+
+	public static async getEndPoints(
+		url: string, username: string, password: string, ignoreSslVerification?: boolean
+	): Promise<IEndPointsResponse> {
+
+		if (!url || !username || !password) {
+			return undefined;
+		}
+
+		url = this.adjustUrl(url);
+		let ep = new EndpointRouterApi(username, password, url);
+		ep.ignoreSslVerification = !!ignoreSslVerification;
+
+		let controllerResponse: IEndPointsResponse = undefined;
+		let controllerError: IControllerError = undefined;
+		let request = <IEndPointsRequest>{
+			url: url,
+			username: username,
+			password: password,
+			method: 'endPointsGet'
+		};
+
+		await ep.endpointsGet().then(result => {
+			controllerResponse = <IEndPointsResponse>{
+				response: result.response as IHttpResponse,
+				endPoints: result.body as IEndPoint[],
+				request
+			};
+		}, error => {
+			if (error) {
+				if ('response' in error) {
+					let err: IEndPointsResponse = error as IEndPointsResponse;
+					let errCode = `${err.response.statusCode || ''}`;
+					let errMessage = err.response.statusMessage;
+					let errUrl = err.response.url;
+					controllerError = <IControllerError>{
+						address: errUrl,
+						code: errCode,
+						errno: errCode,
+						message: errMessage,
+						name: undefined
+					};
+				} else {
+					controllerError = error as IControllerError;
+				}
+				controllerError = Object.assign(controllerError, { request }) as IControllerError;
+			}
+		});
+
+		if (!controllerResponse && !controllerError) {
```

Comment:
There's no need for this with await. Just inline the await response and return the value from it.

Reply:
fixed.